### DISCUSSION TITLE
POST - 400 Bad Request

### DISCUSSION THREAD

**Discussion by PrannavP** [pocketbase/pocketbase#444] [Q&A]
why am i getting bad request error on console while posting data?

**Comment by ganigeorgiev** [comment]
Could you provide an example to what endpoint, body data and how you are sending the request?

_Note that POST is used for create, PATCH for update and DELETE for delete._

If you are using the SDKs, this is done transparently for you, by just calling something like:

```js
// js sdk
await client.records.create("example_collection", {
   name: "test",
});
```

**Reply by briannadoubt** [reply]
I'm getting a 400 when I POST `api/realtime` with a new subscription. Says:
```json
{
  "errorDetails": "clientId: cannot be blank.",
  "errorMessage": "Something went wrong while processing your request."
}
```
However, I'm confident the clientId is being sent in the body, per the [docs](https://pocketbase.io/docs/api-realtime/). The body data is as follows:
```json
{
    "clientId": "aIPLdp1aIZapp757rVZKol944dhdnxXsW7z4ccHn",
    "subscriptions": [
        "test"
    ]
}
```
Seems like something is up with the way my request is being set. Do you have an example network call from the js or dart client that I can reference?

**Reply by ganigeorgiev** [reply]
Hm, maybe a `Content-Type: application/json` header is missing?

For an example, the Dart SDK is calling the base `client.send`:
- https://github.com/pocketbase/dart-sdk/blob/master/lib/src/services/realtime_service.dart#L105
- https://github.com/pocketbase/dart-sdk/blob/master/lib/src/client.dart#L93

The JavaScript SDK is similar but it is using internally `fetch()` for sending the request:
- https://github.com/pocketbase/js-sdk/blob/master/src/Client.ts#L216

**Reply by briannadoubt** [reply]
That was totally it! Realtime is working now. I will push those changes up and start writing some tests! Thank you @ganigeorgiev!
